README.md

```diff
@@ -1,10 +1,162 @@
+and caches. The fields of these configuration objects are a subset of the fields
+required by the `createApp` function.
+
+## The bundle API
+
+### `createApp({ mountPoint, sdk, layout, pages, input, error, cache, local })`
+
+Where:
+
+| fieldname | description |
+| --------- | ----------- |
+| `mountpoint` | A reference to a DOM element to append the app layout to. |
+| `sdk` | An end-user sdk, initialised with the job ID, service ID, and end-user token. |
+| `input` | An object of key value pairs representing input that the job was created with. |
+| `layout` | An object with values as layout templates for the header, the footer, and the summary. |
+| `error` | A template to render errors. |
+| `notFound` | A template to render whenever a page does not exist. |
+| `pages` | A list of configurations for pages to be rendered. |
+| `cache` | A list of specifications for outputs of previous jobs to cache. |
+| `local` | Supplimentary data. |
+
+The `createApp` method is the principle interface of the bundle. By passing it
+various templates, an SDK object, and a flow, this function appends elements to
+the DOM as a child of `mountpoint`. As data becomes available, pages are
+rendered in sequence to acquire more input to progress the job onward to
+completion.
+
+This function requires both an `sdk` object and an `inputs` object, which means
+that the job must already exist before this function is called. It is up to your
+server to perform job creation with initial inputs, and send data to the user
+browser to create the `sdk` object, and subsequently call `createApp`.
+
+#### `sdk`
+
+For documentation regarding the `sdk` object, see [the documentation][sdk]. This
+library requires the _end-user_ form of the `sdk`.
+
+#### `input`
+
+This is exactly the same inputs object which the job is created with.
+
+#### `layout`
+
+Layout consists of a `headers` template, a `footers` template, and a `summary`
+template. The summary template is likely to be the most complex. Each set of
+domain templates included with this module has a layout templates, and it is
+recommended to base your own off of these.
+
+#### `pages`
+
+Pages represent the steps of your automation. Each is composed of the fields:
+
+ - `name`: Then name of the page.
+ - `route`: The route of the page (appears after a `#` in the URL).
+ - `title`: The title of the page, rendered in the layout.
+ - `sections`: An array of section specfications.
+
+Sections are composed of the fields:
+
+ - `name`: The name of the section.
+ - `template`: A function which returns a DOM element or a `DocumentFragment` instance.
+ - `waitFor`: List of data (inputs, outputs, etc.) required in order to render the section.
+
+A template here is a function with parameters in the order: `name`, `options`, `skip`, `sdk`.
+
+ - `name`: The name of the template.
+ - `options`: Various data fields used to render the template.
+ - `skip`: A skip function. ???
```

Comment:
@qubyte can you elaborate before publishing.

Reply:
@kyungeunni I didn't manage to figure this skip function out, but I'm probably missing something obvious. What does it do?

Reply:
@kyungeunni bump!

Reply:
This is a big PR, so I'll update that bit of the README as a followup.